File: strings/both.py
import re 

def both(s):
    arr = s.split(' ')
    result = []
    for x in arr:
        l = len(re.findall('[a-zA-Z]', x))
        n = len(re.findall('\d+', x))
        if l > 0 and n > 0:
            result.append(x)

    return result


str1 = "Emma25 is Data scientist50 and AI Expert"

print(both(str1))

File: strings/both.js


const both = str => {
    let result = [];
    let arr = str.split(' ');

    for (let x in arr) {
        let letters = arr[x].match(/[a-z]/ig);
        let nums = arr[x].match(/\d+/g);

        if (letters && nums && letters.length > 0 && nums.length > 0) {
            result.push(arr[x]);
        }
    }
    return result;
}

let str1 = "Emma25 is Data scientist50 and AI Expert";

console.log(both(str1));